brave/brave-core
# feature(wallet): Compressed NFT integration on iOS

Resolves https://github.com/brave/brave-browser/issues/38738

This PR integrate new APIs to fetch NFT's balance and metadata. Also handles solana compressed NFT transfer. 
However, there are few known UI/UX issues can be addressed later in the future to enhance solana compressed NFT transfer experience. 
1. https://github.com/brave/brave-browser/issues/40336 to display transaction type in string
2. https://github.com/brave/brave-browser/issues/40353 to display compressed NFT information in tx
3. https://github.com/brave/brave-browser/issues/40366 to display correct tx fee



## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues/38738) for my issue
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. make sure you wallet has at least one account owns some NFTs
4. open wallet and go to any screen that could display NFTs e.x portfolio/nfts, nft details, edit visible asset, send token list
5. observe NFT image should be displayed, NFT balance should be correct. NFT owner in NFT details should be correct. 
6. Add a compressed NFT to your Solana account (Compressed NFTs can be obtained via https://drip.haus/ or https://drip.tensor.trade/)
7. try send a compressed nft 
8. a send tx confirmation should pop up
9. tx should be submitted after confirm and eventually the status will updated to confirmed.
Note: Please refer to the known issues that will be addressed later in the PR description above.
